src/backend/app.js

```diff
@@ -21,10 +31,50 @@ app.use(cors());
 
 router.use("/meals", mealsRouter);
 
+//testing knex connection
+router.get("/info", async (req, res) => {
+  const dbResult = await knex.raw("SELECT VERSION()");
+  const row = dbResult[0][0];
+  res.json({ nodeVersion: process.version, mysqlVersion: row["VERSION()"] });
+});
+
+router.get("/future-meals", async (req, res) => {
+  const [rows] = await knex.raw("SELECT * from meal WHERE `when` > CURDATE()");
+  res.json({ rows });
+});
+
+router.get("/past-meals", async (req, res) => {
+  const [rows] = await knex.raw("SELECT * from meal WHERE `when` < CURDATE()");
+  res.json({ rows });
+});
+
+router.get("/all-meals", async (req, res) => {
+  const [rows] = await knex.raw("SELECT * from meal ORDER by id Asc");
+  res.json({ rows });
+});
+
+router.get("/first-meal", async (req, res) => {
+  const [rows] = await knex.raw("SELECT * from meal ORDER by id Asc LIMIT 1");
+  if (rows.length === 0) {
+    res.status(404).send("There are no meals...");
+  } else {
+    res.json({ rows });
+  }
```

Comment:
You can use this `if/else` statement for all the requests.

Reply:
changed